### ISSUE TITLE
Quality of embedded space during training

### ISSUE THREAD

**Issue by DanilZittser** [tensorflow/similarity#211]
Hi to all!

I express my deep gratitude for your product, very perspective!

Please, can you provide example of model training with some distance metric? Especially, if i need only embedding from model without indexing and classification.

```python
model.compile(
    optimizer=tf.keras.optimizers.Adam(learning_rate),
    loss=tfs.losses.MultiSimilarityLoss(distance=distance),
    metrics=<???>,  # which metrics should i use?
)
```

Maybe you can attach some useful links about metrics for embedded space? How i can evaluate quality of this space?

**Comment by owenvallis** [comment]
Hi,

We provide two separate model methods for evaluating the quality of the embedding.
* `SimilarityModel.evaluate_retrieval`: Used to evaluate the quality the embedding given a set of queries and indexed examples. These metrics look at how "well clustered" the embedding space is with respect to the classes.
* `SimilarityModel.evaluate_classification`: Used to evaluate the matching quality given some matching strategy, e.g., match nearest neighbor or a majority vote strategy using K neighbors. These metrics are used to evaluate the classification performance of KNN based classifier.

Additionally, you can pass the classification metrics to `callbacks.EvalCallback` and monitor classification metrics during training.

Some common metrics for evaluating and comparing embedding quality can be found in the `retrieval_metrics` module. It's common to see `recall_at_k` reported in metric learning papers, and the implementation tends to be implemented as "True if Any neighbor in the result matches", and that is the implementation we have in the package. Additionally, I would recommend using `map_at_k` and `bndcg` to get an information retrieval based perspective on the quality of the neighbors.

A good overview of the retrieval metrics can be found in http://arxiv.org/abs/2003.08505 and the following article from Stanford on evaluating ranked retrieval results. https://nlp.stanford.edu/IR-book/html/htmledition/evaluation-of-ranked-retrieval-results-1.html

**Comment by DanilZittser** [comment]
@owenvallis, thanks for quick reply.

I carefully read the API and figured out the metrics in detail. It turned out that all I need is`avg_pos` and `avg_neg` metrics.

```python
model.compile(
    optimizer=tf.keras.optimizers.Adam(learning_rate),
    loss=tfsim.losses.MultiSimilarityLoss(distance=distance),
    metrics=[tfsim.training_metrics.avg_pos(distance=distance), tfsim.training_metrics.avg_neg(distance=distance)],
)
```
